docker/docker-compose.yml

```diff
@@ -10,93 +10,136 @@ etcd:
   command: >
     --listen-client-urls 'http://etcd:4001'
     --advertise-client-urls 'http://etcd:4001'
-#zookeeper:
-#  image: jplock/zookeeper:3.4.5
-#  ports: [ "2181:2181" ]
-mesos:
-  hostname: mesos
+    --initial-cluster-state new
+mesosmaster1:
+  hostname: mesosmaster1
   image: mesosphere/mesos:0.22.0-1.0.ubuntu1404
-  entrypoint: mesos-local
+  entrypoint: mesos-master
   ports: [ "5050:5050" ]
   environment:
-  - MESOS_HOSTNAME=mesos
+  - MESOS_HOSTNAME=mesosmaster1
   - MESOS_PORT=5050
   - MESOS_LOG_DIR=/var/log/mesos
   - MESOS_QUORUM=1
   - MESOS_REGISTRY=in_memory
   - MESOS_WORK_DIR=/var/lib/mesos
-#  - MESOS_ZK=zk://zookeeper:2181/mesos
   links:
   - etcd
   - "ambassador:apiserver"
+mesosslave1:
+  hostname: mesosslave1
+  privileged: true
+  image: mesosphere/kubernetes-mesos-slave:latest
+  ports: [ "5051:5051" ]
+  environment:
+  - MESOS_HOSTNAME=mesosslave1
+  - MESOS_MASTER=mesosmaster1:5050
+  - MESOS_PORT=5051
+  - MESOS_LOG_DIR=/var/log/mesos
+  - MESOS_LOGGING_LEVEL=INFO
+  - MESOS_RESOURCES=cpus:4;mem:1280;disk:25600;ports:[21000-21099]
+  - MESOS_SWITCH_USER=0
+  - MESOS_CONTAINERIZERS=docker,mesos
+  - DOCKER_DAEMON_ARGS="--log-level=error"
+  links:
+  - etcd
+  - mesosmaster1
+  - "ambassador:apiserver"
+  volumes:
+  - /var/run/docker.sock:/var/run/docker.sock
+mesosslave2:
+  hostname: mesosslave2
+  privileged: true
+  image: mesosphere/kubernetes-mesos-slave:latest
+  ports: [ "5052:5052" ]
+  environment:
+  - MESOS_HOSTNAME=mesosslave2
+  - MESOS_MASTER=mesosmaster1:5050
+  - MESOS_PORT=5052
+  - MESOS_LOG_DIR=/var/log/mesos
+  - MESOS_LOGGING_LEVEL=INFO
+  - MESOS_RESOURCES=cpus:4;mem:1280;disk:25600;ports:[22000-22099]
+  - MESOS_SWITCH_USER=0
+  - MESOS_CONTAINERIZERS=docker,mesos
+  - DOCKER_DAEMON_ARGS="--log-level=error"
+  links:
+  - etcd
+  - mesosmaster1
+  - "ambassador:apiserver"
```

Comment:
Why the ambassador:apiserver in the slave? Using the internal 172.x.y.z IP as advertised apiserver name would allow a direct communication between the two.